core/app/[locale]/(default)/cart/_actions/add-shipping-info.ts

```diff
@@ -0,0 +1,126 @@
+import { client } from '~/client';
+import { graphql } from '~/client/graphql';
+import { TAGS } from '~/client/tags';
+
+const UpdateCheckoutShippingConsignmentMutation = graphql(`
+  mutation UpdateCheckoutShippingConsignment($input: UpdateCheckoutShippingConsignmentInput!) {
+    checkout {
+      updateCheckoutShippingConsignment(input: $input) {
+        checkout {
+          entityId
+          shippingConsignments {
+            availableShippingOptions {
+              cost {
+                value
+              }
+              description
+              entityId
+            }
+          }
+        }
+      }
+    }
+  }
+`);
+
+const AddCheckoutShippingConsignmentsMutation = graphql(`
+  mutation AddCheckoutShippingConsignments($input: AddCheckoutShippingConsignmentsInput!) {
+    checkout {
+      addCheckoutShippingConsignments(input: $input) {
+        checkout {
+          entityId
+          shippingConsignments {
+            availableShippingOptions {
+              cost {
+                value
+              }
+              description
+              entityId
+            }
+          }
+        }
+      }
+    }
+  }
+`);
+
+interface Props {
+  checkoutEntityId: string;
+  shippingId?: string;
+  address: {
+    countryCode: string;
+    city?: string;
+    stateOrProvince?: string;
+    postalCode?: string;
+  };
+  lineItems: Array<{ quantity: number; lineItemEntityId: string }>;
+}
+
+export const addShippingInfo = async ({
+  checkoutEntityId,
+  shippingId,
+  address,
+  lineItems,
+}: Props) => {
+  const customerAccessToken = await getSessionCustomerAccessToken();
+
+  let result;
+
+  if (shippingId) {
+    const response = await client.fetch({
+      document: UpdateCheckoutShippingConsignmentMutation,
+      variables: {
+        input: {
+          checkoutEntityId,
+          consignmentEntityId: shippingId,
+          data: {
+            consignment: {
+              address: {
+                ...address,
+                shouldSaveAddress: false,
+              },
+              lineItems,
+            },
+          },
+        },
+      },
+      customerAccessToken,
+      fetchOptions: { cache: 'no-store' },
+    });
+
+    result = response.data.checkout.updateCheckoutShippingConsignment?.checkout;
+  } else {
+    const response = await client.fetch({
+      document: AddCheckoutShippingConsignmentsMutation,
+      variables: {
+        input: {
+          checkoutEntityId,
+          data: {
+            consignments: [
+              {
+                address: {
+                  ...address,
+                  shouldSaveAddress: false,
+                },
+                lineItems,
+              },
+            ],
+          },
+        },
+      },
+      customerAccessToken,
+      fetchOptions: { cache: 'no-store' },
+    });
+
+    result = response.data.checkout.addCheckoutShippingConsignments?.checkout;
+  }
+
+  revalidateTag(TAGS.checkout);
+
+  return result;
```

Comment:
Can we remove the `else` and just utilize early returns here?

Reply:
I agree here 👍  I think early returns are usually preferable. Keeps things clean!